source/core_http_client.c

```diff
@@ -1203,7 +1204,11 @@ static HTTPStatus_t parseHttpResponse( HTTPParsingContext_t * pParsingContext,
     /* This will begin the parsing. Each of the callbacks set in
      * parserSettings will be invoked as parts of the HTTP response are
      * reached. The return code is parsed in #processLlhttpError so is not needed. */
-    ( void ) llhttp_execute( &( pParsingContext->llhttpParser ), parsingStartLoc, parseLen );
+
+    if( llhttp_execute( &( pParsingContext->llhttpParser ), parsingStartLoc, parseLen ) == HPE_PAUSED_UPGRADE )
```

Comment:
It would be clearer to move the llhttp_execute before the if, and then just check the saved return code in the if condition; that way it is clear that that function call has side effects, and is not just for checking the paused upgrade flag